python/orca/src/bigdl/orca/learn/pytorch/torch_runner.py

```diff
@@ -361,17 +367,19 @@ def train_epoch(self, ...):
         from torch.nn.parallel import DistributedDataParallel as DDP
         if isinstance(self.model, DDP):
             with self.model.join():
-                self._train_loop(iterator, metric_meters, callbacks)
+                self._train_loop(iterator, metric_meters, callbacks, max_steps)
         else:
-            self._train_loop(iterator, metric_meters, callbacks)
+            self._train_loop(iterator, metric_meters, callbacks, max_steps)
 
         self.call_hook(callbacks=callbacks, fn_name="on_lr_adjust")
 
         return metric_meters.summary(sync_stats=self.sync_stats,
                                      dist_backend=self.dist_backend)
 
-    def _train_loop(self, iterator, metric_meters, callbacks):
+    def _train_loop(self, iterator, metric_meters, callbacks, max_steps=None):
         for batch_idx, batch in enumerate(iterator):
+            if max_steps is not None and self.global_step >= max_steps:
```

Comment:
I prefer to integrate this into a EarlyStopping Callback after this merges? @hkvision We shall take it carefully to add any new arg in torchrunner, for they are really hard to be maintained.

Reply:
earlystop is to stop with metric criteria. I don't think we need to integrate training with max steps to earlystop as max step means training only specific step without any condition. We may refer to pytorch lightning, max steps is a trainer options. EarlyStop is callback to stop when some condition met.

Reply:
You can keep this in this pr, and we may refactor this just like what we do in TqdmCallback? @hkvision  I think training only specific steps is also one kind of condition. What about we set up a flag like `runner.stop`, when any condition reaches we change `runner.stop` to True and runner checks it and exits in EarlyStop hooks.